x/nft/types/nft.go

```diff
@@ -41,36 +47,46 @@ func NewBaseNFT(ID uint64, owner sdk.AccAddress, tokenURI, description, image, n
 	}
 }
 
+// GetID returns the ID of the token
+func (bnft BaseNFT) GetID() uint64 { return bnft.ID }
+
+// GetOwner returns the account address that owns the NFT
+func (bnft BaseNFT) GetOwner() sdk.AccAddress { return bnft.Owner }
+
+// GetName returns the name of the token
+func (bnft BaseNFT) GetName() string { return bnft.Name }
+
+// GetDescription returns the unique description of the NFT
+func (bnft BaseNFT) GetDescription() string { return bnft.Description }
+
+// GetImage returns the image path of the NFT
+func (bnft BaseNFT) GetImage() string { return bnft.Image }
+
+// GetTokenURI returns the path to optional extra properties
+func (bnft BaseNFT) GetTokenURI() string { return bnft.TokenURI }
+
 // EditMetadata edits metadata of an nft
-func (nft *BaseNFT) EditMetadata(editName, editDescription, editImage, editTokenURI bool,
-	name, description, image, tokenURI string) {
-	if editName {
-		nft.Name = name
-	}
-	if editDescription {
-		nft.Description = description
-	}
-	if editImage {
-		nft.Image = image
-	}
-	if editTokenURI {
-		nft.TokenURI = tokenURI
-	}
+func (bnft BaseNFT) EditMetadata(name, description, image, tokenURI string) {
+	(&bnft).Name = name
+	(&bnft).Description = description
+	(&bnft).Image = image
+	(&bnft).TokenURI = tokenURI
 }
 
-func (nft BaseNFT) String() string {
-	return fmt.Sprintf(`	ID: 					%d
-	Owner:        			%s
-  	Name:         			%s
-  	Description: 			%s
-  	Image:        			%s
-	TokenURI:   			%s`,
-		nft.ID,
-		nft.Owner,
-		nft.Name,
-		nft.Description,
-		nft.Image,
-		nft.TokenURI,
+func (bnft BaseNFT) String() string {
+	return fmt.Sprintf(`ID: 					%d
+	Owner:        %s
+  Name:         %s
+  Description: 	%s
+  Image:        %s
+	TokenURI:   	%s
+	`,
+		bnft.ID,
+		bnft.Owner,
+		bnft.Name,
+		bnft.Description,
+		bnft.Image,
+		bnft.TokenURI,
```

Comment:
@fedekunze is this the intentional indentation?
```
ID: 					%d
	Owner:        %s
  Name:         %s
  Description: 	%s
  Image:        %s
	TokenURI:   	%s
	`,
		bnft.ID,
		bnft.Owner,
		bnft.Name,
		bnft.Description,
		bnft.Image,
		bnft.TokenURI,
```

Reply:
no, idk why my indentation is broken lol